app/http.php

```diff
@@ -327,4 +378,50 @@
     }
 });
 
+function getDomains(Database $dbForConsole, &$lastSyncUpdate, &$domains, $pools)
+{
+    go(function () use ($dbForConsole, &$lastSyncUpdate, &$domains, $pools) {
+        Timer::tick(DOMAIN_SYNC_TIMER * 1000, function () use ($dbForConsole, &$domains, &$lastSyncUpdate, $pools) {
+            $time = DateTime::now();
+            $timerStart = \microtime(true);
+            var_dump($domains);
+            $limit = 1000;
+            $sum = $limit;
+            $total = 0;
+            $latestDocument = null;
+
+            while ($sum === $limit) {
+                $queries = [Query::limit($limit)];
+                if ($latestDocument !== null) {
+                    $queries[] =  Query::cursorAfter($latestDocument);
+                }
+                if ($lastSyncUpdate != null) {
+                    $queries[] = Query::greaterThanEqual('$updatedAt', $lastSyncUpdate);
+                }
+                $queries[] = Query::equal('resourceType', ['function']);
+
+                $results = Authorization::skip(fn () =>  $dbForConsole->find('rules', $queries));
+
+                $sum = count($results);
+                $total = $total + $sum;
+                foreach ($results as $document) {
+                    $domain = $document->getAttribute('domain');
+                    if (str_contains($domain, '.appwrite.global')) {
+                        continue;
+                    }
+                    $domains[$domain] = true;
+                }
+                $latestDocument = !empty(array_key_last($results)) ? $results[array_key_last($results)] : null;
+            }
+
+            $lastSyncUpdate = $time;
+            $timerEnd = \microtime(true);
+
+            $pools->reclaim();
+
+            Console::log("Sync domains tick: {$total} domains were updated in " . ($timerEnd - $timerStart) . " seconds");
```

Comment:
Hmm, this could end up as spam. Can we print only if there was actually some new rules?